src/store/accounts/account.hooks.ts

```diff
@@ -30,6 +36,25 @@ export function useCurrentAccountAvailableStxBalance() {
 export function useAccountActivity() {
   return useAtomValue(currentAccountTransactionsState);
 }
+
+const stxTransferFromContractCall = (txs: AddressTransactionWithTransfers[]) => {
+  const contractCallTxs = txs.filter(tx => 'contract_call' in tx.tx);
+  return contractCallTxs
+    .map(tx =>
+      tx.stx_transfers.map(transfer => ({
+        ...tx.tx,
+        tx_id: tx.tx.tx_id,
+        sender_address: transfer.sender,
+        token_transfer: { amount: transfer.amount },
+        tx_type: 'token_transfer',
+      }))
+    )
```

Comment:
I'd strongly advise against this. This breaks the type system as many of the other properties of `MempoolTransaction | Transaction` aren't actually there. This is effectively tricking the type system, which will be really confusing to any developer who has to touch this later (and could well result in runtime errors not caught at compile time).

We should think of another way to format the list of txs that doesn't involve creating a "matching" type